editor/src/clj/editor/graphics.clj

```diff
@@ -426,24 +481,30 @@
 
 (defn- editable-attribute-info? [attribute-info]
   (case (:semantic-type attribute-info)
-    (:semantic-type-position :semantic-type-texcoord :semantic-type-page-index :semantic-type-normal) false
+    (:semantic-type-position :semantic-type-texcoord :semantic-type-page-index :semantic-type-normal :semantic-type-world-matrix :semantic-type-normal-matrix) false
     nil false
     true))
 
+(defn contains-semantic-type? [attributes semantic-type]
+  (some #(= semantic-type (:semantic-type %)) attributes))
+
 (defn- attribute-property-type [attribute]
   (case (:semantic-type attribute)
     :semantic-type-color types/Color
-    (case (int (:element-count attribute))
-      1 g/Num
-      2 types/Vec2
-      3 types/Vec3
-      4 types/Vec4)))
-
-(defn- attribute-expected-element-count [{:keys [element-count semantic-type] :as _attribute}]
-  {:pre [(integer? element-count)]}
+    (case (:vector-type attribute)
+      :vector-type-scalar g/Num
+      :vector-type-vec2 types/Vec2
+      :vector-type-vec3 types/Vec3
+      :vector-type-vec4 types/Vec4
+      :vector-type-mat2 types/Vec4
+      :vector-type-mat3 types/Vec4
+      :vector-type-mat4 types/Vec4)))
```

Comment:
We didn't have matrix types before for attributes and since we can edit them as vec4's in the material editor I figured we'd want the same functionality for custom attributes.